numpy/core/src/multiarray/mapping.c

```diff
@@ -228,8 +228,7 @@ unpack_scalar(PyObject *index, PyObject **result, npy_intp NPY_UNUSED(result_n))
 NPY_NO_EXPORT npy_intp
 unpack_indices(PyObject *index, PyObject **result, npy_intp result_n)
 {
-    npy_intp n, i;
-    npy_bool commit_to_unpack;
+    npy_intp n;
 
     /* Fast route for passing a tuple */
     if (PyTuple_CheckExact(index)) {
```

Comment:
I think the function below `unpack_tuple` (unlike the code suggests, probably my bad from many years ago), will just work for subclasses.  So, I would suggest to just change this:
```suggestion
    if (PyTuple_Check(index)) {
```
At least in C-Python, tuple subclass checking is quick.  The only check down there that I think may make sense to bump would be `!PyLong_CheckExact` because `arr[0]` is the one thing that might even notice it.

That said, I doubt it will notice this, the only reason would be that the `tp_flags` slot is cold (not in cache) and frankly, we probably need that chunk of memory anyway later :).

Reply:
To be clear:  That then allows to delete most of what follows.

Reply:
Done. I agree this is simpler!